tests/test_result_conversion.rs

```diff
@@ -0,0 +1,45 @@
+//! Testing https://github.com/PyO3/pyo3/issues/1106. A result type that
```

Comment:
Could please place this in `test_various` or so?

Reply:
Sure. I think `test_various` shouldn't be a thing because that way it's easier to understand the test, but I'll follow your style.